leftwm-core/src/utils/state_socket.rs

```diff
@@ -23,7 +23,7 @@ pub struct StateSocket {
 impl Drop for StateSocket {
     fn drop(&mut self) {
         assert!(
-            std::thread::panicking() || !self.listener.is_some(),
+            std::thread::panicking() || self.listener.is_none(),
```

Comment:
Just out of curiosity, did clippy complain about this, or did you just notice this by your self?

Reply:
Ye a new clippy

Reply:
Ah, being new makes sense, as it didn't complain about this, when I pushed #646